tools/autoupdate_app_sources/rest_api.py

```diff
@@ -176,3 +178,18 @@ def releases(self) -> list[dict[str, Any]]:
     def url_for_ref(self, ref: str, ref_type: RefType) -> str:
         """Get a URL for a ref."""
         return f"{self.forge_root}/{self.project_path}/archive/{ref}.tar.gz"
+
+
+class DownloadPageAPI:
+    def __init__(self, upstream: str) -> None:
+        self.web_page = upstream
+
+    def get_web_page_links(self) -> dict[str, str]:
+        r = requests.get(self.web_page)
+        r.raise_for_status()
+        soup = BeautifulSoup(r.text, features="lxml")
```

Comment:
lxml and bs4 are missing from requirements.txt

Reply:
Fixed.